src/main/java/com/google/sps/user/User.java

```diff
@@ -0,0 +1,90 @@
+package com.google.sps.user;
+
+import org.jetbrains.annotations.Nullable;
+
+/** Stores the data related to one user. */
+public class User {
+    private final String id, email;
+    private final boolean publicPortfolio;
+    @Nullable 
+    private final String name, selfIntroduction, imgUrl;
+
+    public static class Builder {
+        // Required.
+        private final String id, email;
+        // Optional.
+        private String name, selfIntroduction, imgUrl;
+        private boolean publicPortfolio = false;
+
+        public Builder(String id, String email) {
+            this.id = id;
+            this.email = email;
+        }
+
+        public Builder setName(String name) {
+            this.name = name;
+            return this;
+        }
+
+        public Builder addImgUrl(String imgUrl) {
+            this.imgUrl = imgUrl;
+            return this;
+        }
+
+        public Builder addSelfIntroduction(String selfIntroduction) {
+            this.selfIntroduction = selfIntroduction;
+            return this;
+        }
+
+        public Builder setPublicPortfolio() {
```

Comment:
I suggest to use a param to set. It is more readable and flexible.

Reply:
Added a parameter.